Bump to JDK 11

oguzhanunlu opened the pull request.

chuwy commented on modules/kafka/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/sources/KafkaSource.scala:
```diff
@@ -133,7 +133,7 @@ class KafkaSource private (
   private def createConsumer(brokers: String, groupId: String): KafkaConsumer[String, Array[Byte]] = {
     val properties = createProperties(brokers, groupId)
-    properties.putAll(kafkaConfig.consumerConf.getOrElse(Map()).asJava)
+    kafkaConfig.consumerConf.getOrElse(Map()).foreach { case (k, v) => properties.setProperty(k, v) }
```
Does it actually work? I tried it a week ago and tests were failng with different errors.

oguzhanunlu replied:
Initially I hit this bug https://github.com/scala/bug/issues/10418 and wanted to work-around it by this line. Now I've hit another [error](https://github.com/snowplow/enrich/pull/360/checks#step:9:2101) on `Class org.apache.kafka.common.serialization.StringSerializer could not be found` and trying to solve it.

oguzhanunlu replied:
> tests were failng with different errors.

Do you refer to unit tests? Or custom tests you performed manually? I'd like to learn more about those failing tests. @chuwy

chuwy replied:
I refer to exactly that error you're getting now. I couldn't fix it.

oguzhanunlu replied:
I just removed that test and CI looks happy now

oguzhanunlu commented:
@benjben @chuwy if you agree with the changes, I'd like to publish rc1 and see it in action

chuwy commented on build.sbt:
```diff
@@ -178,13 +178,4 @@ lazy val beam =
     )
     .enablePlugins(JavaAppPackaging, DockerPlugin, BuildInfoPlugin)
 
-lazy val integrationTests = project
```
But we didn't delete the actual module?

oguzhanunlu replied:
This commit deletes `modules/integration-tests`, did you mean something else? @chuwy

chuwy replied:
No, that's what I meant - sorry didn't notice somehow.

chuwy commented:
Please create a ticket, @oguzhanunlu explaining changes and force-push the ticket-less commit.

And don't forget to resolve the conflict.

benjben commented on modules/beam/src/test/scala/com.snowplowanalytics.snowplow.enrich.beam/enrichments/YauaaEnrichmentSpec.scala:
```diff
@@ -36,7 +36,8 @@ object YauaaEnrichmentSpec {
     "event" -> "page_ping",
-    "derived_contexts" -> json"""{"schema":"iglu:com.snowplowanalytics.snowplow/contexts/jsonschema/1-0-1","data":[{"schema":"iglu:nl.basjes/yauaa_context/jsonschema/1-0-1","data":{"deviceBrand":"Unknown","deviceName":"Desktop","operatingSystemVersionMajor":"7","layoutEngineNameVersion":"Gecko 12.0","operatingSystemNameVersion":"Windows 7","layoutEngineBuild":"20100101","layoutEngineNameVersionMa ...
+    "derived_contexts" -> json"""{"schema":"iglu:com.snowplowanalytics.snowplow/contexts/jsonschema/1-0-1","data":[{"schema":"iglu:nl.basjes/yauaa_context/jsonschema/1-0-1","data":{"deviceBrand":"Unknown","deviceName":"Desktop","operatingSystemVersionMajor":"7","layoutEngineNameVersion":"Gecko 12.0","operatingSystemNameVersion":"Windows 7","layoutEngineBuild":"20100101","layoutEngineNameVersionMa ...
+  )
```
Not sure what this has to do with Java 11

benjben commented on modules/common/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/outputs/EnrichedEventSpec.scala:
```diff
@@ -112,42 +112,42 @@ class EnrichedEventSpec extends Specification {
-      testField(_.pp_xoffset_max = new JInteger(0), _.pp_xoffset_max)
-      testField(_.pp_yoffset_min = new JInteger(0), _.pp_yoffset_min)
-      testField(_.pp_yoffset_max = new JInteger(0), _.pp_yoffset_max)
+      testField(_.ti_quantity = JInteger.valueOf(0), _.ti_quantity)
+      testField(_.pp_xoffset_min = JInteger.valueOf(0), _.pp_xoffset_min)
+      testField(_.pp_xoffset_max = JInteger.valueOf(0), _.pp_xoffset_max)
+      testField(_.pp_yoffset_min = JInteger.valueOf(0), _.pp_yoffset_min)
+      testField(_.pp_yoffset_max = JInteger.valueOf(0), _.pp_yoffset_max)
```
All of these were taken care of in https://github.com/snowplow/enrich/commit/86536a539e4b5dd659e1d9d07bd76f69e3bef454

I should have created a PR just for this and merge right away. I didn't because it was preventing me from compiling but actually I could have created a new branch just for this and rebase my branch on top of it waiting for the merge. I will do that next time. Sorry!

oguzhanunlu replied:
No worries @benjben ! Thanks for the empathy!

benjben reviewed (commented):
Thanks @oguzhanunlu !

I believe that many OSS users use Kafka as a message queue. Would it require a lot of work to get the integration tests to work? If not we might want to invest the time.

oguzhanunlu commented:
Thanks for the comments @chuwy , I just addressed your feedback. @benjben , it took me several hours and I couldn't make that test compile at jdk 11. Feel free to give it a try, but I doubt it is worth the effort, re-writing from scratch would take shorter, if we'd like to offer tests for kafka module.

chuwy reviewed (approved):
Just gave some feedback over Slack. Otherwise - 👍

benjben reviewed (commented):
LGTM but let's see with Emily if we can remove Kafka integration tests.

benjben reviewed (approved):
Sorry it was my misunderstanding, we are still testing that enrich can read from and write to Kafka.

chuwy commented:
Feel free to merge, @oguzhanunlu

oguzhanunlu commented:
Thanks @chuwy , proceeding
